src/data_race.rs

```diff
@@ -437,6 +437,47 @@ impl MemoryCellClocks {
 /// Evaluation context extensions.
 impl<'mir, 'tcx: 'mir> EvalContextExt<'mir, 'tcx> for MiriEvalContext<'mir, 'tcx> {}
 pub trait EvalContextExt<'mir, 'tcx: 'mir>: MiriEvalContextExt<'mir, 'tcx> {
+    /// Temporarily allow data-races to occur, this should only be
+    /// used if either one of the appropriate `validate_atomic` functions
+    /// will be called to treat a memory access as atomic or if the memory
+    /// being accessed should be treated as internal state, that cannot be
+    /// accessed by the interpreted program.
```

Comment:
This documentation comment should be updated now to reflect the cleanup of env vars.